packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -635,6 +635,18 @@ class IOSDevice extends Device {
     _logReaders[app] = logReader;
   }
 
+  @visibleForTesting
+  Future<bool> isDeveloperModeEnabled() async {
+    final bool enabled = await _iosDeploy.isDeveloperModeEnabled(
+        deviceId: id
+    );
+    if (!enabled) {
+      _logger.printError('To use $name for development, enable Developer Mode in Settings → Privacy & Security.');
```

Comment:
This error output is an unexpected side-effect of essentially a getter, what if the caller doesn't think it's an error, it just wants to check to handle something differently based on what mode it's in?  Instead, print errors/call toolExits in the caller where it actually knows that this is a Bad Thing.